src/test/java/marquez/api/resources/DatasetResourceTest.java

```diff
@@ -15,77 +15,173 @@
+import java.util.Optional;
 import javax.ws.rs.core.Response;
+import marquez.UnitTests;
+import marquez.api.exceptions.NamespaceNotFoundException;
+import marquez.api.mappers.DatasetMapper;
+import marquez.api.mappers.DatasetResponseMapper;
+import marquez.api.models.DatasetRequest;
 import marquez.api.models.DatasetResponse;
 import marquez.api.models.DatasetsResponse;
+import marquez.common.models.ConnectionUrl;
+import marquez.common.models.DatasetName;
 import marquez.common.models.DatasetUrn;
+import marquez.common.models.DatasourceName;
+import marquez.common.models.DatasourceUrn;
+import marquez.common.models.Description;
 import marquez.common.models.NamespaceName;
 import marquez.service.DatasetService;
 import marquez.service.NamespaceService;
 import marquez.service.exceptions.MarquezServiceException;
 import marquez.service.models.Dataset;
-import marquez.service.models.Generator;
+import org.junit.Before;
 import org.junit.Test;
+import org.junit.experimental.categories.Category;
 
+@Category(UnitTests.class)
 public class DatasetResourceTest {
-  private static final DatasetUrn DATASET_URN = Generator.genDatasetUrn();
-  private static final Instant CREATED_AT = Instant.now();
-  private static final NamespaceName NAMESPACE_NAME = NamespaceName.fromString("test");
-  private static final Integer LIMIT = 100;
-  private static final Integer OFFSET = 0;
-
-  private final NamespaceService mockNamespaceService = mock(NamespaceService.class);
-  private final DatasetService mockDatasetService = mock(DatasetService.class);
-  private final DatasetResource datasetResource =
-      new DatasetResource(mockNamespaceService, mockDatasetService);
-
-  @Test(expected = NullPointerException.class)
-  public void testNamespaceServiceNull() {
+  private static final int LIMIT = 100;
+  private static final int OFFSET = 0;
+
+  private static final NamespaceName NAMESPACE_NAME = newNamespaceName();
+  private static final ConnectionUrl CONNECTION_URL = newConnectionUrl();
+  private static final DatasourceName DATASOURCE_NAME = newDatasourceName();
+  private static final DatasourceUrn DATASOURCE_URN =
+      DatasourceUrn.from(CONNECTION_URL, DATASOURCE_NAME);
+  private static final DatasetName DATASET_NAME = newDatasetName();
+  private static final DatasetUrn DATASET_URN = DatasetUrn.from(DATASOURCE_NAME, DATASET_NAME);
+  private static final Description DESCRIPTION = newDescription();
+  private static final Dataset DATASET = newDatasetWith(DATASET_NAME, DATASET_URN, DESCRIPTION);
+  private static final DatasetRequest DATASET_REQUEST =
+      new DatasetRequest(DATASET_NAME, DATASOURCE_URN, DESCRIPTION);
+  private static final Dataset NEW_DATASET = DatasetMapper.map(DATASET_REQUEST);
+
+  private NamespaceService namespaceService;
+  private DatasetService datasetService;
+  private DatasetResource datasetResource;
+
+  @Before
+  public void setUp() {
+    namespaceService = mock(NamespaceService.class);
```

Comment:
You only really want to set these up once per class, not once per method. Please use `@BeforeClass`.

Reply:
I normally use `@BeforeClass` for more expensive setup steps we only want to run once per test suite. With `@Before`, we can reinstantiate our mocks per tests. What are the benefits of using `@BeforeClass` instead?

Reply:
The benefit is you can build something once and re-use (by clearing mocks before or after every test), instead of rebuilding it every time. I agree that it's not computationally intensive to re-instantiate a mock, but I think it's a good practice to do things only when necessary, since that thinking carries over to other test fixtures that could be more expensive.

Reply:
The benefit is you can build something once and re-use (by clearing mocks before or after every test), instead of rebuilding it every time. I agree that it's not computationally intensive to re-instantiate a mock, but I think it's a good practice to do things only when necessary, since that thinking carries over to other test fixtures that could be more expensive.

Reply:
It's safer to just use `@Before` and get a fresh one. `@BeforeClass` is better for suite-level setup / teardown or avoiding something too expensive to do for every test method. In this case, instantiating a mock `Namespace` is super cheap, so no harm doing it in `@Before`, and no need to worry about cleaning things up manually.